modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/ValueProviderModelValidator.java

```diff
@@ -45,48 +54,89 @@
+        if (processedValueProviders.contains(otherValueProviderImplentation)) {
+          return;
+        }
+        if (valueProviderId.equals(otherValueProviderId)) {
+          problemsReporter.addError(new Problem(valueProviderInformation.ownerModel,
+                                                format("Parameter '%s' from the %s '%s' uses an implementation of ValueProviders"
+                                                    + " [%s] with id '%s'. There is another implementations of ValueProviders"
+                                                    + " [%s] that uses the same id. ValueProviders id must be unique.",
+                                                       valueProviderInformation.parameterModel.getName(),
+                                                       getComponentModelTypeName(ownerModel), getModelName(ownerModel),
+                                                       valueProviderImplentation, valueProviderId,
+                                                       otherValueProviderImplentation)));
+        }
+      });
+    });
   }
 
-  private void validateModel(ParameterizedModel model, ProblemsReporter problemsReporter, boolean supportsConnectionsAndConfigs) {
+  private void validateModel(ParameterizedModel model, ProblemsReporter problemsReporter, boolean supportsConnectionsAndConfigs,
+                             Map<String, ValueProviderInformation> valueProviders) {
     ReflectionCache reflectionCache = new ReflectionCache();
     model.getAllParameterModels()
         .forEach(param -> param
             .getModelProperty(ValueProviderFactoryModelProperty.class)
             .ifPresent(modelProperty -> validateOptionsResolver(param, modelProperty, model, problemsReporter,
-                                                                supportsConnectionsAndConfigs, reflectionCache)));
+                                                                supportsConnectionsAndConfigs, reflectionCache, valueProviders)));
   }
 
   private void validateOptionsResolver(ParameterModel param, ValueProviderFactoryModelProperty modelProperty,
                                        ParameterizedModel model, ProblemsReporter problemsReporter,
-                                       boolean supportsConnectionsAndConfigs, ReflectionCache reflectionCache) {
+                                       boolean supportsConnectionsAndConfigs, ReflectionCache reflectionCache,
+                                       Map<String, ValueProviderInformation> valueProviders) {
     Class<? extends ValueProvider> valueProvider = modelProperty.getValueProvider();
     String providerName = valueProvider.getSimpleName();
+    Optional<ValueProviderModel> valueProviderModel = param.getValueProviderModel();
+    if (!valueProviderModel.isPresent()) {
+      throw new IllegalStateException(format("Parameter %s from %s with name %s has should have a ValueProviderModel associated.",
```

Comment:
This should never happen. The parameterModel will always have the valueProviderModel if the modelProperty is present, I did not want to get the value from the optional without checking anyway, so that if this breaks by any chance, we have a good error message. I can make it fail only at compile time.